boards/BMS/Inc/App/App_Accumulator.h

```diff
@@ -1,185 +1,48 @@
     NUM_OF_ACCUMULATOR_SEGMENTS,
 } AccumulatorSegments_E;
 
-struct Accumulator;
-
 /**
  * Allocate and initialize an accumulator.
- * @param configure_cell_monitors A function to configure all cell monitors for
- * the given accumulator.
- * @param read_cell_voltages A function to read cell voltages from all cell
- * monitors for the given accumulator.
- *
- * @param get_min_cell_voltage A function that returns the absolute minimum cell
- * voltage of the accumulator.
- * @param get_max_cell_voltage A function that returns the absolute maximum cell
- * voltage of the accumulator.
- * @param get_average_cell_voltage A function that returns the average cell
- * voltage for the accumulator.
- * @param get_pack_voltage A function that returns the accumulator pack voltage.
- * @param get_segment_0_voltage A function that returns the voltage of the 0th
- * accumulator segment.
- * @param get_segment_1_voltage A function that returns the voltage of the 1st
- * accumulator segment.
- * @param get_segment_2_voltage A function that returns the voltage of the 2nd
- * accumulator segment.
- * @param get_segment_3_voltage A function that returns the voltage of the 3rd
- * accumulator segment.
- * @param get_segment_4_voltage A function that returns the voltage of the 4th
- * accumulator segment.
- * @param get_segment_5_voltage A function that returns the voltage of the 5th
- * accumulator segment.
- *
- * @param min_cell_voltage The minimum cell voltage for the given accumulator.
- * @param max_cell_voltage The maximum cell voltage for the given accumulator.
- * @param min_segment_voltage The minimum segment voltage for the given
- * accumulator.
- * @param max_segment_voltage The maximum segment voltage for the given
- * accumulator.
- * @param min_pack_voltage The minimum pack voltage for the given accumulator.
- * @param max_pack_voltage The maximum pack voltage for the given accumulator.
- *
- * @return A pointer to the created accumulator, whose ownership is given to
- * the caller.
+ * @param configure_cell_monitors A function that can be called to configure
+ * cell monitors
+ * @param start_voltage_conv A function that can be called to trigger voltage
+ * conversions
+ * @param read_cell_voltages A function that can be called to read cell voltages
+ * @param get_min_cell_voltage A function to get the minimum cell voltage (V)
+ * @param get_max_cell_voltage A function to get the maximum cell voltage (V)
+ * @param get_segment_voltage A function to get a given segment voltage (V)
+ * @param get_pack_voltage A function to get the pack voltage (V)
  */
 struct Accumulator *App_Accumulator_Create(
-    ExitCode (*configure_cell_monitors)(void),
-    ExitCode (*read_cell_voltages)(void),
-
+    bool (*configure_cell_monitors)(void),
+    bool (*start_voltage_conv)(void),
+    bool (*read_cell_voltages)(void),
+    void (*get_min_cell_location)(uint8_t *, uint8_t *),
+    void (*get_max_cell_location)(uint8_t *, uint8_t *),
```

Comment:
What are your thoughts on including the names of the function pointer arguments? It's not clear what the two uint8_t ptrs are supposed to represent

Reply:
i think we can refer to the io function that we currently pass in right now